src/unitxt/operators.py

```diff
@@ -202,13 +201,12 @@ def process(
 
     def get_mapped_value(self, instance, key, mapper, val):
         val_as_str = str(val)  # make sure the value is a string
-        if self.strict and (val_as_str not in mapper):
+        if val_as_str in mapper:
+            return mapper[val_as_str]
```

Comment:
Why is deep copy not needed?  What happens if we map a value to a list and then change the list?

Reply:
Hi @yoavkatz , you probably mean if this happens inside `metric` (in `ApplyMetric` operator). We noticed that only `PipelineMetric` contains a shameless stepping over `prediction` and `references` (no matter through which step) as part of its `preprocess_steps` or `postprocess_steps`, so we save `prediction` and `references` as step0  of the `preprocess_steps` of `PipelineMetric`, and restore upon leaving. So `prediction` and `references` are saved against any stupidity - copy list and break there, or copy a dictionary and break there.

Reply:
Hi @yoavkatz , The description of the damage implies `ListFieldValues` operator, not `MapInstanceValues` that your comment sits on its lines.  Isn't it?